packages/mask/src/components/DataSource/usePersonaAgainstSNSConnectStatus.ts

```diff
@@ -0,0 +1,34 @@
+import type { PersonaInformation, NextIDPlatform } from '@masknet/shared-base'
+import { useAsyncRetry } from 'react-use'
+import { useMemo } from 'react'
+import Services from '../../extension/service'
+import { activatedSocialNetworkUI } from '../../social-network'
+import { useLastRecognizedIdentity } from './useActivatedUI'
+import { useMyPersonas } from './useMyPersonas'
+import { useSetupGuideStatusState } from './useNextID'
+
+export function usePersonaAgainstSNSConnectStatus() {
+    const ui = activatedSocialNetworkUI
+    const platform = ui.configuration.nextIDConfig?.platform as NextIDPlatform
+    const lastState = useSetupGuideStatusState()
+    const lastRecognized = useLastRecognizedIdentity()
+    const username = useMemo(() => {
+        return lastState.username || lastRecognized.identifier?.userId
+    }, [lastState, lastRecognized])
+    const personas = useMyPersonas()
```

Comment:
looks like all of those data are only used in the async function. I think you should move the async function to the top level and subscribe to the events to upgrade/retry the async function.

Reply:
🤔